massa-pool-worker/src/endorsement_pool.rs

```diff
@@ -128,10 +129,13 @@ impl EndorsementPool {
                     added.insert(endo.id);
                 }
 
-                // And send endorsements to Denunciation Factory
+                // And send endorsements to Denunciation pool
                 match DenunciationPrecursor::try_from(endo) {
```

Comment:
problem: the endorsement pool is destined to contain endorsements for inclusion in new blocks. For example it ignores endorsements that are before the latest final blocks (see line "// endorsement expired: ignore" above) but those can still be denounced and the denunciation pool should be notified about them.

Conceptually, the denunciation pool should not be notified of endorsements by the endorsement pool. However, at every place where we send endorsements to the endorsement pool (eg. protocol, factory), we should also notify the denunication pool.

Reply:
So why not sending the endorsement before the mentioned check?

Reply:
Described in DM

Reply:
done